pkg/ibi/imageclusterinstall_test.go

```diff
@@ -0,0 +1,772 @@
+		testBuilder := generateImageClusterInstallBuilder()
+
+		testBuilder.WithExtraManifests(testCase.extramanifest)
+		assert.Equal(t, testCase.expectedErrorMsg, testBuilder.errorMsg)
+
+		if testCase.expectedErrorMsg == "" {
+			assert.Equal(t, testCase.extramanifest, testBuilder.Definition.Spec.ExtraManifestsRefs[0].Name)
+		}
+	}
+}
+func TestImageClusterInstallWithMachineNetwork(t *testing.T) {
+	testCases := []struct {
+		machineNetwork   string
+		expectedErrorMsg string
+	}{
+		{
+			machineNetwork:   "192.168.0.0/24",
+			expectedErrorMsg: "",
+		},
+		{
+			machineNetwork:   "fd2e:6f44:5dd8::/64",
+			expectedErrorMsg: "",
+		},
+		{
+			machineNetwork:   "192.168.0.0/255.255.255.0",
+			expectedErrorMsg: "imageclusterinstall machinenetwork incorrectly formatted",
+		},
+		{
+			machineNetwork:   "fd2e:6f44:5dd8::",
+			expectedErrorMsg: "imageclusterinstall machinenetwork incorrectly formatted",
+		},
+		{
+			machineNetwork:   "",
+			expectedErrorMsg: "imageclusterinstall machinenetwork incorrectly formatted",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateImageClusterInstallBuilder()
+
+		testBuilder.WithMachineNetwork(testCase.machineNetwork)
+		assert.Equal(t, testCase.expectedErrorMsg, testBuilder.errorMsg)
+
+		if testCase.expectedErrorMsg == "" {
+			assert.Equal(t, testCase.machineNetwork, testBuilder.Definition.Spec.MachineNetwork)
+		}
+	}
+}
+
+func TestImageClusterInstallWithSSHKey(t *testing.T) {
+	testCases := []struct {
+		sshkey           string
+		expectedErrorMsg string
+	}{
+		{
+			sshkey:           "mysecretsshkey",
+			expectedErrorMsg: "",
+		},
+		{
+			sshkey:           "",
+			expectedErrorMsg: "imageclusterinstall sshkey cannot be empty",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateImageClusterInstallBuilder()
+
+		testBuilder.WithSSHKey(testCase.sshkey)
+		assert.Equal(t, testCase.expectedErrorMsg, testBuilder.errorMsg)
+
+		if testCase.expectedErrorMsg == "" {
+			assert.Equal(t, testCase.sshkey, testBuilder.Definition.Spec.SSHKey)
+		}
+	}
+}
+func TestImageClusterInstallWithVersion(t *testing.T) {
+	testCases := []struct {
+		version          string
+		expectedErrorMsg string
+	}{
+		{
+			version:          "mysecretsshkey",
+			expectedErrorMsg: "",
+		},
+		{
+			version:          "",
+			expectedErrorMsg: "imageclusterinstall version cannot be empty",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateImageClusterInstallBuilder()
+
+		testBuilder.WithVersion(testCase.version)
+		assert.Equal(t, testCase.expectedErrorMsg, testBuilder.errorMsg)
+
+		if testCase.expectedErrorMsg == "" {
+			assert.Equal(t, testCase.version, testBuilder.Definition.Spec.Version)
+		}
+	}
+}
+func TestImageClusterInstallGetCompletedCondition(t *testing.T) {
+	testCases := []struct {
+		status                   ibiv1alpha1.ImageClusterInstallStatus
+		expectedConditionMessage string
+		expectedError            error
+	}{
+		{
```

Comment:
Do we want another testcase with wrong status?

Reply:
The method just returns a condition object based on the type of the condition. I have test cases for when the condition type appears and doesn't appear in the status.